libsycl/src/ld-version-script.txt

```diff
@@ -0,0 +1,19 @@
+{
+  /* Do not use extern "C++" matcher for C++ functions,                     */
+  /* because vtable and typeinfo symbols make extern "C++" patterns more    */
+  /* complicated than patterns against mangled names.                       */
+  /*                                                                        */
+  /* With extern "C++" we have to match for "vtable for sycl::foo", but */
+  /* not match for "vtable for std::__internal<sycl::foo>".             */
```

Comment:
```suggestion
  /* Symbols to be exported are selected based on mangled names rather than   */
  /* the demangled names provided by the `extern "C++"` matcher because it is */
  /* easy to express "export everything defined in the sycl namespace" using  */
  /* the former. Matching demangled names is more complicated in the presence */
  /* of examples like:                                                        */
  /*   "vtable for sycl::foo"                  (should be exported)           */
  /*   "vtable for std::__internal<sycl::foo>" (should not be exported)       */
```

Reply:
will update, thanks

Reply:
updated in https://github.com/llvm/llvm-project/pull/144372/commits/6e473009602cdf0ce5cb2fb2f4fe27acc60125d2#diff-012c1c75642ff105a8083560c92d539c11a7d62cbe4946820528c77ba8875270